packages/carousel/src/index.js

```diff
@@ -0,0 +1,228 @@
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      items: this.props.children,
+      internalCurrentItemIndex: this.props.currentItemIndex || 0,
+      currentItemTranslate: 0,
+      isSwiping: false
+    }
+
+    this.handleSwipingLeft = this.handleSwipingLeft.bind(this)
+    this.handleSwipingRight = this.handleSwipingRight.bind(this)
+    this.handleSwipedLeft = this.handleSwipedLeft.bind(this)
+    this.handleSwipedRight = this.handleSwipedRight.bind(this)
+    this.updateSelectedItem = this.updateSelectedItem.bind(this)
+    this.mapItems = this.mapItems.bind(this)
+    this.mapDots = this.mapDots.bind(this)
+  }
+
+  handleSwipingLeft(e, absX) {
+    const swipedPercentage = (absX / window.innerWidth) * 100
+    const swipeLeft = swipedPercentage <= 100 ? swipedPercentage : 100
+    this.setState(previousState => ({
+      currentItemTranslate: 0 - swipeLeft,
+      isSwiping: true
+    }))
+  }
+
+  handleSwipingRight(e, absX) {
+    const swipedPercentage = (absX / window.innerWidth) * 100
+    const swipeRight = swipedPercentage <= 100 ? swipedPercentage : 100
+    this.setState(previousState => ({
+      currentItemTranslate: 0 + swipeRight,
+      isSwiping: true
+    }))
+  }
+
+  handleSwipedLeft(e, absX) {
+    const index =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+    const nextItemIndex = (index + 1) % this.state.items.length
+
+    this.setState({
+      internalCurrentItemIndex: nextItemIndex,
+      currentItemTranslate: 0,
+      isSwiping: false
+    })
+  }
+
+  handleSwipedRight(e, absX) {
+    const index =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+    const previousItemIndex =
+      (index + this.state.items.length - 1) % this.state.items.length
+
+    this.setState({
+      internalCurrentItemIndex: previousItemIndex,
+      currentItemTranslate: 0,
+      isSwiping: false
+    })
+  }
+
+  updateSelectedItem(selectedItemIndex) {
+    this.setState({
+      internalCurrentItemIndex: selectedItemIndex
+    })
+  }
+
+  mapItems() {
+    const { items } = this.state
+
+    const currentItemIndex =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+
+    const itemsToShow = []
+    items.forEach((item, index) => {
+      if (index <= currentItemIndex + 2) {
+        itemsToShow.push(
+          <span key={`item-${index}`}>
+            <PositionItem>{item}</PositionItem>
+          </span>
+        )
+      }
+    })
+
+    return itemsToShow
+  }
+
+  mapDots() {
+    const currentItemIndex =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+
+    return this.state.items.map((item, mapIndex) => (
+      <Dot
+        key={mapIndex}
+        mx={2}
+        mt={2}
+        bg={mapIndex === currentItemIndex ? 'blue' : 'borderGray'}
+        onClick={this.updateSelectedItem.bind(this, mapIndex)}
+      />
+    ))
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
This is a deprecated component method – generally you shouldn't need to derive state from props changes. Guessing this could be handled in the render method at first glance